bindings/Fortran/CMakeLists.txt

```diff
@@ -61,6 +61,33 @@ else()
    )
 endif()
 
+# feature check: real(kind=16) types are not supported with PGI
+function(_Fortran_has_real16 success)
+  if(NOT ${success})
+    set(scratch_directory
+      ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/fortran)
+    set(test_file ${scratch_directory}/cmake_real16_test.f90)
+    file(WRITE ${test_file}
+      "program fortran_real16_hello\n"
+      "  real(kind=16) :: x\n"
+      "end\n")
+    try_compile(${success} ${scratch_directory} ${test_file})
+    if(${success})
+        message(STATUS "Fortran: real(kind=16) supported")
+        set(${success} TRUE PARENT_SCOPE)
+    else()
+        set(${success} FALSE PARENT_SCOPE)
+        message(STATUS "Fortran: real(kind=16) NOT supported")
+    endif()
+  endif()
+endfunction()
+
+set(ADIOS2_HAVE_Fortran_REAL16 FALSE)
+_Fortran_has_real16(ADIOS2_HAVE_Fortran_REAL16)
+if(Fortran_HAS_REAL16)
+  target_compile_definitions(adios2_f PUBLIC ADIOS2_HAVE_Fortran_REAL16)
```

Comment:
@chuckatkins I do need some flags here to be able to use `#ifdef` in Fortran source files.

I could not figure out why e.g. `Test.Bindings.Fortran.Remove` can already do this and I cannot here, did I miss a property?

Reply:
Oddly, you need to capitalize the postfix...   .F90 instead of .f90.

Reply:
> Oddly, you need to capitalize the postfix... .F90 instead of .f90.

Fortran convention is for `foo.F90` to invoke the compiler with the preprocessor while `foo.f90` gets invoked without it.  While it's not "standardized", it is effectively given that the convention is used by all the compilers we support.

Reply:
Thank you so much, today I learned :D

Reply:
FWIW, here's a few references to the convention:

- PGI: https://www.pgroup.com/resources/docs/19.10/x86/pgi-user-guide/index.htm#fn-conv-input
- GNU: https://gcc.gnu.org/onlinedocs/gfortran/Preprocessing-and-conditional-compilation.html
- Intel: https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-understanding-file-extensions
- XL: https://www.ibm.com/support/knowledgecenter/SSAT4T_16.1.1/com.ibm.xlf1611.lelinux.doc/compiler_ref/cpp.html
- Cray: https://pubs.cray.com/content/S-3901/9.1/cray-fortran-reference-manual/source-preprocessing

Basically they all say to use an uppercase extension for automatic pre-processing.  We could go through the trouble of trying to explicitly add the per-compiler pre-processor flag, which CMake might already know about, but it's much easier to just use the "unofficial standard but convention in practice" method of an uppercase file extension.

Reply:
Cool, yes agreed the ending makes sense for our target compilers. Thanks for the background info!

Reply:
@chuckatkins I can locally compile with GCC-8/GFortran-8 but CI does not agree. Did I oversee something?